### ISSUE TITLE
Client table  "content" (screenshot) not usable

### ISSUE THREAD

**Issue by hari-rangarajan** [awesomeWM/awesome#581]
The icon works fine, but the content throws an error "cairo.Surface: no 'width'". The code returns both the icon and content as a cairo surface. Is the usage incorrect here?

	local iconbox = wibox.widget.imagebox()
	iconbox:set_image(c.icon)

	local clientimagebox = wibox.widget.imagebox()
	clientimagebox:set_image(c.content)

**Comment by Elv13** [comment]
Use `gears.surface(c.content)`, this should work

**Comment by psychon** [comment]
The imagebox only works with cairo ImageSurface instances. `c.content` returns an XCBSurface...
Try the following:
```lua
local cairo = require("lgi").cairo
local content = gears.surface(c.content)
local cr = cairo.Context(content)
local x, y, w, h = cr:clip_extents()
local img = cairo.ImageSurface.create(cairo.Format.ARGB32, w - x, h - y)
cr = cairo.Context(img)
cr:set_source_surface(content, 0, 0)
cr.operator = cairo.Operator.SOURCE
cr:paint()

local clientimagebox = wibox.widget.imagebox(img)
```

**Comment by psychon** [comment]
We now have `gears.surface.duplicate_surface()` making the above even easier. Also, I don't think fixing `imagebox` to make your code work is a good idea, because `c.content` isn't a screenshot but a "live reference" to the window's content, so weird things will happen if the window's content changes.

TL;DR: Closing since `gears.surface.duplicate_surface(gears.surface(c.content))` should get things to work.
